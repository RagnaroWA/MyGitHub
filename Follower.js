import React, { Component } from 'react';
import { BottomNavigation, Text } from 'react-native-paper';
import { AppRegistry, StyleSheet, View, Image, Linking, ScrollView, TouchableOpacity, TouchableWithoutFeedback, Dimensions, AsyncStorage } from 'react-native';
import { Button, List, withTheme, type Theme } from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialIcons';
import axios from 'axios';
import { Divider } from 'react-native-elements';
import ProfileModal from './ProfileModal.js';

class Follower extends Component {
  constructor(props) {
    super(props);

    Dimensions.addEventListener('change', () => {
      this.setState({
        orientation: isPortrait() ? 'portrait' : 'landscape'
      });
      this.forceUpdate();
    });

    const isPortrait = () => {
      const dim = Dimensions.get('screen');
      return dim.height >= dim.width;
    }

    this.state = {
      login : "",
      id : -1,
      node_id : "",
      avatar_url : "",
      bio : "",
      blog : "",
      created_at : "",
      email : "",
      followers_url : "",
      following_url : "",
      name : "",
      public_repos : "",
      repos_url : "",
      url : "",
      followers : [],
      modal_visible: false,
      modal_follower: null,
    };
    this.replaceNbsps = this.replaceNbsps.bind(this);
    this.openUrl = this.openUrl.bind(this);

    this.changeStyle = this.changeStyle.bind(this);
    var {height, width} = Dimensions.get('window');
    this.state['styles'] = this.changeStyle(width, height);
    this.openModal = this.openModal.bind(this);
    this.setModalVisible = this.setModalVisible.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.addFollowing = this.addFollowing.bind(this);
  }

  addFollowing(follower) {
    let token = this.state['token'];
    putFollowing = async () => {
      let result_following = await axios.put("https://api.github.com/user/following/" + follower['login'], {}, token);
      if(result_following !== null) {
        console.log("Start get following !!!");
        this.props.children.getFollowing();
      }
    };
    putFollowing();
  }

  setModalVisible(visible) {
    this.setState({modal_visible: visible});
  }

  closeModal() {
    this.setState({
      modal_visible: false,
    });
  }

  openModal(follower) {
    let obj = {"follower": follower, "closeFunc": this.closeModal}
    this.setState({
      modal_visible: true,
      modal_follower: obj,
    });
  }

  replaceNbsps(str) {
    let re = new RegExp(String.fromCharCode(160), "g");
    return str.replace(re, " ");
  }

  openUrl(url) {
		Linking.canOpenURL(url).then(supported => {
      		if (supported) {
        		Linking.openURL(url);
      		}
      		else {
        		console.log("Don't know how to open URI: " + url);
      		}
    	});
	}

  changeStyle(width, height) {
    var styles = StyleSheet.create({
      container : {
    		marginTop: height * 0.02,
    		backgroundColor: '#F6F8FA',
    		width: width,
        flexGrow: 1,
        flex: 2
    	},

      titlebutton : {
        marginTop: height * 0.04,
        marginBottom: height * 0.03,
        borderRadius: 19,
        borderColor: "black",
        borderWidth: 1,
        alignSelf: 'flex-start',
        marginLeft: 'auto',
        marginRight: 'auto',
      },

      title : {
        fontWeight: 'bold',
        fontSize: 20,
        marginTop: height * 0.01,
        marginLeft: width * 0.05,
        marginRight: width * 0.05,
        marginBottom: height * 0.01,
        textAlign: 'center',
        color: 'black',
      },

      divider : {
        backgroundColor: 'black',
        height: 1,
      },

      follower : {
        marginTop: 5,
        width : width,
      },

      followerlogin : {
    	  fontSize: 16,
    	  marginLeft: 10,
    	  marginTop: 7,
      	fontFamily: 'Gill Sans',
      },

      company : {
        marginTop: 7,
        marginLeft: 'auto',
        marginRight: 3,
        fontSize: 17,
      },

      companyname : {
        marginTop: 6,
        marginRight: 20,
        fontSize: 16,
      },

      avatar : {
        width: 40,
        height: 40,
        marginLeft: 15,
        marginBottom: 10,
        marginTop: 10,
        borderRadius: 20,
        borderColor: 'black',
        borderWidth: 1,
      },

      followername : {
      	fontSize: 15,
      	marginLeft: 15,
      	marginTop: 10,
      	fontFamily: 'Gill Sans',
        transform: [
          {
            translateY : 12.5
          }
        ],
      },

      cancelButton : {
        marginTop: 15,
        marginLeft: 'auto',
        marginRight: 20,
        fontSize: 17,
      },

      cancelIcon : {
        fontSize: 22,
      },

      bio : {
      	fontSize: 14,
      	marginLeft: 5,
      	marginBottom: 10,
      	marginTop: 0,
      	fontFamily: 'Gill Sans',
      },

      bioicon : {
      	fontSize: 16,
      	marginLeft: 15,
      	marginBottom: 10,
    	  marginTop: 0,
      },

      empty : {
        height: height * 0.05,
      },
    });
    return styles;
  }

  componentWillMount() {
    if(this.props['children'] != undefined) {
      let data = this.props['children']['App'];
      this.setState({
        login : data['login'],
        id : data['id'],
        node_id : data['node_id'],
        avatar_url : data['avatar_url'],
        bio : data['bio'],
        blog : data['blog'],
        created_at : data['created_at'],
        email : data['email'],
        following : data['following'],
        followers_url : data['followers_url'],
        following_url : data['following_url'],
        name : data['name'],
        public_repos : data['public_repos'],
        repos_url : data['repos_url'],
        url : data['url'],
        company : data['company'],
        token : data['token'],
      });
      getFollower = async (url) => {
  			let result_follower = await axios.get(url);
  			let followers = result_follower.data;
        // console.log(result_follower);
  			for(let i = 0; i<followers.length; i ++) {
  				let follower = {};
  				follower['id'] = followers[i]['id'];
  				follower['avatar_url'] = followers[i]['avatar_url'];
  				follower['type'] = followers[i]['type'];
  				follower['html_url'] = followers[i]['html_url'];
  				follower['login'] = followers[i]['login'];
  				follower['id'] = followers[i]['id'];
          follower['api_url'] = followers[i]['url'];
          let follower_data = await axios.get(follower['api_url']);
          console.log(follower_data);
          follower['name'] = follower_data.data['name'];
          if(follower['name'] === null || follower['name'] === "") {
            follower['name'] = follower['login'];
          }
          follower['company'] = follower_data.data['company'];
          if(follower['company'] === null) {
            follower['company'] = "N/A";
          }
          if(follower['company'].length > 20) {
            follower['company'] = follower['company'].substring(0, 17) + "...";
          }
          follower['bio'] = follower_data.data['bio'];
          if(follower['bio'] === "" || follower['bio'] === null) {
            follower['bio'] = "N/A";
          }
          if(follower['bio'].length > 40) {
            follower['bio'] = follower['bio'].substring(0, 37) + "...";
          }
          follower['blog'] = follower_data.data['blog'];
          if(follower['blog'] === null || follower['blog'] === "") {
            follower['blog'] = follower_data.data['html_url'];
          }
          follower['created_at'] = follower_data.data['created_at'];
          follower['followers'] = follower_data.data['followers'];
          follower['following'] = follower_data.data['following'];
          follower['public_repos'] = follower_data.data['public_repos'];
          follower['html_url'] = follower_data.data['html_url'];
          let createDate = new Date(Date.parse(follower['created_at']));
          let month = createDate.getUTCMonth() + 1;
          let day = createDate.getUTCDate();
          let year = createDate.getUTCFullYear();
          let createDateFinal = createDate.toISOString().split('T')[0];
          follower['created_at'] = createDateFinal;
  				this.state['followers'].push(follower);
  			}
        let followers_save = this.state['followers'];
        console.log(this.state['token']['auth']['username']);
        console.log(followers_save);

        const saveFollowerInfo = async followers_save => {
          try {
            await AsyncStorage.setItem(this.state['token']['auth']['username']+"followers", JSON.stringify(followers_save));
          } catch (error) {
            console.log(error.message);
          }
        }
        saveFollowerInfo(followers_save);

        const getFollowerInfo = async () => {
          let followers_load = {};
          try {
            followers_load = await AsyncStorage.getItem(this.state['token']['auth']['username']+"followers", (value) => {
              JSON.parse(value);
            });
            console.log(followers_load);
          } catch (error) {
            // Error retrieving data
            console.log(error.message);
          }
        }
        getFollowerInfo();
  			this.forceUpdate();
	    };
	    getFollower(data['url']+"/followers");
      console.log(data['url']+"/followers");
	    this.forceUpdate();
  	}
  }

    render() {
      var {height, width} = Dimensions.get('window');
      if(this.state['orientation'] === "portrait") {
        this.state['styles'] = this.changeStyle(width, height);
      }
      else {
        this.state['styles'] = this.changeStyle(width, height);
      }
      const styles = this.state['styles'];
      return (
        <ScrollView style={styles.container}>
          {
            this.state['url'].length > 0 ? (
              <TouchableOpacity style={styles.titlebutton} onPress={this.openUrl.bind(this, this.state['url']+"?tab=followers")}>
                <Text style={styles.title}>{"Followers".toUpperCase()}</Text>
              </TouchableOpacity>
            ) : (
              <TouchableOpacity style={styles.titlebutton}>
                <Text style={styles.title}>{this.state['name']}</Text>
              </TouchableOpacity>
            )
          }
          <Divider style={styles.divider} />
          {
              this.state.modal_visible ?
             <ProfileModal>{this.state['modal_follower']}</ProfileModal> :

             this.state['followers'].map((follower) => (
               <View key={follower['id']}>
                 <View style={styles.follower}>
                   <View style={{flexDirection: 'row'}}>
                     <TouchableOpacity onPress={() => this.openModal(follower)}>
                       <Text style={styles.followerlogin}>
                         {follower['login'].toUpperCase()}
                       </Text>
                     </TouchableOpacity>
                     <Icon name="work" color="black" style={styles.company} ></Icon>
                     <Text style={styles.companyname} >{follower['company']}</Text>
                   </View>

                   <View style={{flexDirection: 'row'}}>
                     <Image style={styles.avatar} source={{uri: follower['avatar_url']}}></Image>
                     <Text style={styles.followername}>Name: {follower['name']}</Text>
                       <TouchableOpacity style={styles.cancelButton} onPress={() => this.addFollowing(follower)}>
                         <Icon name="done" color="black" style={styles.cancelIcon} ></Icon>
                       </TouchableOpacity>
                   </View>
                 </View>
                 <View style={{flexDirection: 'row'}}>
                   <Icon name="description" style={styles.bioicon} color="black" ></Icon>
                   <Text style={styles.bio}>{follower['bio']}</Text>
                 </View>
                 <Divider style={styles.divider} />
               </View>
             ))
          }
          <View style={styles.empty}></View>
        </ScrollView>
      );
    }
}
export default Follower;
