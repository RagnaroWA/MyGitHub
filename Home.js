import React, { Component } from 'react';
import { BottomNavigation, Text } from 'react-native-paper';
import { AppRegistry, StyleSheet, View, Image, Linking, ScrollView } from 'react-native';
import { Button, List, withTheme, type Theme } from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialIcons';
import axios from 'axios';
import Dimensions from 'Dimensions';

class Home extends Component {
  constructor(props) {
    super(props);

    Dimensions.addEventListener('change', () => {
      this.setState({
        orientation: isPortrait() ? 'portrait' : 'landscape'
      });
      this.forceUpdate();
    });

    const isPortrait = () => {
      const dim = Dimensions.get('screen');
      return dim.height >= dim.width;
    }

    this.state = {
      login : "",
      id : -1,
      node_id : "",
      avatar_url : "",
      bio : "",
      blog : "",
      created_at : "",
      email : "",
      followers : -1,
      following : -1,
      followers_url : "",
      following_url : "",
      name : "",
      public_repos : "",
      repos_url : "",
      url : "",
    };
    this.replaceNbsps = this.replaceNbsps.bind(this);
    this.state['orientation'] = isPortrait() ? 'portrait' : 'landscape';

    this.changeStyle = this.changeStyle.bind(this);
    var {height, width} = Dimensions.get('window');
    this.state['styles'] = this.changeStyle(width, height);
  }

  replaceNbsps(str) {
    let re = new RegExp(String.fromCharCode(160), "g");
    return str.replace(re, " ");
  }

  changeStyle(width, height) {
    const styles = StyleSheet.create({
      centerText1 : {
        fontWeight: 'bold',
        fontSize: 22,
        marginTop: height * 0.08,
        textAlign: 'center'
      },

      centerText2 : {
        fontSize: 19,
        marginTop: height * 0.008,
        textAlign: 'center',
        color: '#4f4f4f'
      },

      background : {
        backgroundColor: '#F6F8FA',
        flexGrow: 1,
        flex: 2
      },

      website : {
        fontSize: 13,
        color: 'black'
      },

      avatar : {
        width: width * 0.4,
        height: width * 0.4,
        marginTop: height * 0.01,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.4 * 0.5
          }
        ],
        borderRadius: 20,
      },

      company : {
        fontWeight: 'bold',
        fontSize: 16,
        marginTop: height * 0.03,
        textAlign: 'center',
      },

      bio : {
        fontSize: 18,
        marginTop: 10,
        textAlign: 'center',
        color: 'black',
        width: width * 0.8,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.8 * 0.5
          }
        ],
        marginBottom: 40,
      },

      follower : {
        width: width * 0.55,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.55 * 0.5
          }
        ],
        marginTop: height * -0.025,
        fontSize: 12,
        borderRadius: 20,
        borderColor: "black",
        borderWidth: 1,
        marginBottom: 20,
      },

      following : {
        marginTop: height * 0.0,
        width: width * 0.55,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.55 * 0.5
          }
        ],
        fontSize: 12,
        borderRadius: 20,
        borderColor: "black",
        borderWidth: 1,
        marginBottom: 20,
      },

      repo : {
        marginTop: height * 0.0,
        width: width * 0.55,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.55 * 0.5
          }
        ],
        fontSize: 12,
        borderRadius: 20,
        borderColor: "black",
        borderWidth: 1,
        marginBottom: 20,
      },

      create : {
        fontSize: 15,
        marginTop: height * 0.03,
        textAlign: 'center',
        color: '#4f4f4f'
      },

      empty : {
        height: height * 0.05,
      },
    });
    return styles;
  }

  componentWillMount() {
    // axios.defaults.headers.common['Authorization'] = auth;
    // getUsers = async () => {
    //   let result = await axios.get('https://api.github.com/users/' + 'RagnaroWA');
    //   let data = result.data;
    //   // console.log(data);
    //   let createDate = new Date(Date.parse(data['created_at']));
    //   let month = createDate.getUTCMonth() + 1;
    //   let day = createDate.getUTCDate();
    //   let year = createDate.getUTCFullYear();
    //   let createDateFinal = createDate.toISOString().split('T')[0];
    //   this.setState({
    //     login : data['login'],
    //     id : data['id'],
    //     node_id : data['node_id'],
    //     avatar_url : data['avatar_url'],
    //     bio : this.replaceNbsps(data['bio']),
    //     blog : data['blog'],
    //     created_at : createDateFinal,
    //     email : data['email'],
    //     followers : data['followers'],
    //     following : data['following'],
    //     followers_url : data['followers_url'],
    //     following_url : data['following_url'],
    //     name : data['name'],
    //     public_repos : data['public_repos'],
    //     repos_url : data['repos_url'],
    //     url : data['url'],
    //     company : data['company'],
    //   });
    // };
    // getUsers();
    // console.log(this.props.children);
    // this.forceUpdate();
  }

  componentDidMount() {
    if(this.props['children'] != undefined) {
      // console.log(this.props['children']['App']);
      let data = this.props['children']['App'];
      if(data['bio'] === null) {
        data['bio'] = 'N/A';
      }
      if(data['company'] === null) {
        data['company'] = 'N/A';
      }
      console.log(data);
      this.setState({
        login : data['login'],
        id : data['id'],
        node_id : data['node_id'],
        avatar_url : data['avatar_url'],
        bio : data['bio'],
        blog : data['blog'],
        created_at : data['created_at'],
        email : data['email'],
        followers : data['followers'],
        following : data['following'],
        followers_url : data['followers_url'],
        following_url : data['following_url'],
        name : data['name'],
        public_repos : data['public_repos'],
        repos_url : data['repos_url'],
        url : data['url'],
        company : data['company'],
      });
    }
  }

  render() {
    let numberFollowers = `Followers: ${this.state['followers'].toString()}`;
    let numberFollowings = `Followings: ${this.state['following'].toString()}`;
    let numberRepos = `Repositories: ${this.state['public_repos'].toString()}`;
    let email = this.state['email'];
    if (email == null) {
      email = "zzhan147@illinois.edu"
    }
    var {height, width} = Dimensions.get('window');
    if(this.state['orientation'] === "portrait") {
      this.state['styles'] = this.changeStyle(width, height);
    }
    else {
      this.state['styles'] = this.changeStyle(width, height);
    }
    const styles = this.state['styles'];
      return (
        <ScrollView style={styles.background}>
          <Text style={styles.centerText1}>{this.state['name']}</Text>
          <Text style={styles.centerText2}>{this.state['login']}</Text>
          <Button style={styles.website} color="#4f4f4f" onPress={ ()=>{ Linking.openURL('mailto:'+email)}}>{email}</Button>
          <Button style={styles.website} color="black" onPress={ ()=>{ Linking.openURL(this.state['blog'])}}>{this.state['blog']}</Button>
          <Image style={styles.avatar} source={{uri: this.state['avatar_url']}}></Image>
          <Text style={styles.company}>{this.state['company']}</Text>
          <Text style={styles.bio}>{this.state['bio']}</Text>
          <Button style={styles.follower} icon="person-add" onPress={() => this.props.children.changeFollower()} color="black">{numberFollowers}</Button>
          <Button style={styles.following} icon="person" onPress={() => this.props.children.changeFollowing()} color="black">{numberFollowings}</Button>
          <Button style={styles.repo} icon="folder" onPress={() => this.props.children.changeRepo()} color="black">{numberRepos}</Button>
          <Text style={styles.create}>Created on: {this.state['created_at']}</Text>
          <View style={styles.empty}></View>
        </ScrollView>
      );
  }
}

export default Home;
