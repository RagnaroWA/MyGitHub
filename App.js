import React, { Component } from 'react';
import { BottomNavigation } from 'react-native-paper';
import { AppRegistry, StyleSheet, View, Modal, TouchableHighlight, Text, Alert, TextInput, Image, Dimensions, AsyncStorage } from 'react-native';
import axios from 'axios';
import Home from './Home.js';
import Repo from './Repo.js';
import Follower from './Follower.js';
import Following from './Following.js';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { Button, List, withTheme, type Theme } from 'react-native-paper';

var HomeRoute = () => <Home>{this.changeFollower}</Home>;
var FollowerRoute = () => <Follower>Follower</Follower>;
var FollowingRoute = () => <Following>Following</Following>;
var RepoRoute = () => <Repo>Repo</Repo>;

class App extends Component {
  constructor(props) {
    super(props);

    Dimensions.addEventListener('change', () => {
      this.setState({
        orientation: isPortrait() ? 'portrait' : 'landscape'
      });
      this.forceUpdate();
    });

    const isPortrait = () => {
      const dim = Dimensions.get('screen');
      return dim.height >= dim.width;
    }

    this.state = {
      index: 0,
      routes: [
        { key: 'home', title: 'Home', icon: 'account-circle' },
        { key: 'followers', title: 'Follower', icon: 'person-add' },
        { key: 'followings', title: 'Following', icon: 'person'},
        { key: 'repos', title: 'Repository', icon: 'folder' },
      ],
      userlogin: false,
      scene: this._renderScene,
      username_login: '',
      username_password: '',
    };
    this.changeFollower = this.changeFollower.bind(this);
    this.changeFollowing = this.changeFollowing.bind(this);
    this.changeRepo = this.changeRepo.bind(this);
    this.getFollowing = this.getFollowing.bind(this);
    this.child = React.createRef();


    this.replaceNbsps = this.replaceNbsps.bind(this);
    this.callApi = this.callApi.bind(this);

    this.state['orientation'] = isPortrait() ? 'portrait' : 'landscape';
    this.changeStyle = this.changeStyle.bind(this);

    let {height, width} = Dimensions.get('window');
    this.state['styles'] = this.changeStyle(width, height);
  }

  getFollowing() {
    this.child.current.updateFollower(this.state['url']+"/following");
    console.log(this.state['url']+"/following");
  }

  changeStyle(width, height) {
    const styles = StyleSheet.create({
      modalContainer : {
        justifyContent: 'center',
        flexGrow: 1,
        flex: 2
      },

      modalavatar : {
        height: width * 0.35,
        width: width * 0.35,
        marginBottom: 20,
        transform: [
          {
            translateX : 1 * width * 0.65 * 0.5
          }
        ],
        borderRadius: 20,
      },

      putusername : {
        height: 40,
        width: width * 0.7,
        transform: [
          {
            translateX : 1 * width * 0.3 * 0.5
          }
        ],
        borderColor: 'black',
        marginBottom: 20,
        borderWidth: 2,
        fontSize: 22,
        textAlign: 'center',
      },

      putuserpassword : {
        height: 40,
        width: width * 0.7,
        transform: [
          {
            translateX : 1 * width * 0.3 * 0.5
          }
        ],
        borderColor: 'black',
        marginBottom: 20,
        borderWidth: 2,
        fontSize: 22,
        textAlign: 'center',
      },

      loginButton : {
        marginTop: height * 0.015,
        width: width * 0.4,
        left: width * 0.5,
        transform: [
          {
            translateX : -1 * width * 0.4 * 0.5
          }
        ],
        marginTop: height * 0.005,
        fontSize: 13,
        borderRadius: 20,
        borderColor: "black",
        borderWidth: 2
      },
    });
    return styles;
  }

  setModalVisible(visible) {
    this.setState({userlogin: visible});
  }

  replaceNbsps(str) {
    let re = new RegExp(String.fromCharCode(160), "g");
    return str.replace(/\xA0/g,'');
  }

  changeFollower() {
    this.setState({index : 1})
  }

  changeFollowing() {
    this.setState({index : 2})
  }

  changeRepo() {
    this.setState({index : 3})
  }

  checkUser() {
    let token = {
        auth: {
            username: this.state.username_login,
            password: this.state.username_password,
        }
    };
    getValid = async () => {
      try {
        let result = await axios.get('https://api.github.com/users/' + this.state.username_login, {}, token);
        let data = result.data;
        console.log(result);
        this.setState({
          userlogin: true,
        });
        this.state['userlogin'] = true;
        this.callApi();
      }
      catch (error){
        console.log(error);
        Alert.alert("Wrong Username or Password.");
      }
    }
    getValid();
  }

  callApi() {
    // auth = {
    //   "username": 'RagnaroWA',
    //   "token": 'b0bcc9fc100ab25f5c29413f21b01d598d147879'
    // };
    // axios.defaults.headers.common['Authorization'] = auth;
    var token = {
        auth: {
            username: this.state.username_login,
            password: this.state.username_password,
        }
    };
    getUsers = async () => {
      let result = await axios.get('https://api.github.com/users/' + this.state.username_login, {}, token);
      let data = result.data;
      let createDate = new Date(Date.parse(data['created_at']));
      let month = createDate.getUTCMonth() + 1;
      let day = createDate.getUTCDate();
      let year = createDate.getUTCFullYear();
      let createDateFinal = createDate.toISOString().split('T')[0];
      if(data['bio'] === null) {
        data['bio'] = 'N/A';
      }
      console.log(data['bio']);
      if(data['company'] === null) {
        data['company'] = 'N/A';
      }
      this.setState({
        login : data['login'],
        id : data['id'],
        node_id : data['node_id'],
        avatar_url : data['avatar_url'],
        bio : this.replaceNbsps(data['bio']),
        blog : data['blog'],
        created_at : createDateFinal,
        email : data['email'],
        followers : data['followers'],
        following : data['following'],
        followers_url : data['followers_url'],
        following_url : data['following_url'],
        name : data['name'],
        public_repos : data['public_repos'],
        repos_url : data['repos_url'],
        url : data['url'],
        company : data['company'],
        token : token,
      });

      const userInfo = this.state;
      const saveUserInfo = async userInfo => {
        try {
          await AsyncStorage.setItem(this.state.username_login, JSON.stringify(userInfo));
        } catch (error) {
          console.log(error.message);
        }
      }
      saveUserInfo(data);

      const getLoginId = async () => {
        let userInfo = {};
        try {
          userInfo = await AsyncStorage.getItem(this.state.username_login, (value) => {
            JSON.parse(value);
          });
          console.log(userInfo);
        } catch (error) {
          // Error retrieving data
          console.log(error.message);
        }
      }
      getLoginId();

      let props = {
        changeFollower: this.changeFollower,
        changeFollowing: this.changeFollowing,
        changeRepo: this.changeRepo,
        getFollowing: this.getFollowing,
        "App": this.state,
      };
      HomeRoute = () => <Home>{props}</Home>;
      RepoRoute = () => <Repo>{props}</Repo>;
      FollowingRoute = () => <Following ref={this.child}>{props}</Following>
      FollowerRoute = () => <Follower>{props}</Follower>
      _renderScene = BottomNavigation.SceneMap({
          home: HomeRoute,
          followers: FollowerRoute,
          followings: FollowingRoute,
          repos: RepoRoute,
        });
        this.setState({
          scene: _renderScene,
        });
    };
    getUsers();
    _renderScene = BottomNavigation.SceneMap({
        home: HomeRoute,
        followers: FollowerRoute,
        followings: FollowingRoute,
        repos: RepoRoute,
      });
      this.setState({
        scene: _renderScene,
      });
  }

  _handleIndexChange = index => this.setState({ index });

  _renderScene = BottomNavigation.SceneMap({
      home: HomeRoute,
      followers: FollowerRoute,
      followings: FollowingRoute,
      repos: RepoRoute,
    });

  render() {
    var {height, width} = Dimensions.get('window');
    console.log(height);
    if(this.state['orientation'] === "portrait") {
      this.state['styles'] = this.changeStyle(width, height);
    }
    else {
      this.state['styles'] = this.changeStyle(height, width);
    }
    let styles = this.state['styles'];
    if(this.state['userlogin'] === false) {
        return (
          <View>
            <Modal
              animationType="slide"
              transparent={false}
              visible={true}
              onRequestClose={() => {
                Alert.alert('Modal has been closed.');
              }}>
              <View style={styles.modalContainer}>
                <View>
                  <Image
                    style={styles.modalavatar}
                    source={require('./icon.png')}
                  />
                  <TextInput
                    style={styles.putusername}
                    onChangeText={(username_login) => this.setState({username_login})}
                    value={this.state.username_login}
                  />
                  <TextInput
                    style={styles.putuserpassword}
                    onChangeText={(username_password) => this.setState({username_password})}
                    value={this.state.username_password}
                    secureTextEntry={true}
                  />
                <Button style={styles.loginButton} color="black" onPress={ ()=>{ this.checkUser(); }}>GitHub Login</Button>
                </View>
              </View>
            </Modal>
          </View>
      );
    }
    else {
      return (
          <BottomNavigation
            navigationState={this.state}
            onIndexChange={this._handleIndexChange}
            renderScene={this.state.scene}
            barStyle={{backgroundColor: "#23292D"}}
          />
      );
    }
  }
}

export default App;
