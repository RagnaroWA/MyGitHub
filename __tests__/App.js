/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Home from '../Home';
import Follower from '../Follower';
import Following from '../Following';
import Repo from '../Repo';
import ProfileModal from '../ProfileModal';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
