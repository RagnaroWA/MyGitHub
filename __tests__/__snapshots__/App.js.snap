// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <View
      style={
        Object {
          "flex": 2,
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "testUri": "../../../icon.png",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 262.5,
              "marginBottom": 20,
              "transform": Array [
                Object {
                  "translateX": 243.75,
                },
              ],
              "width": 262.5,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "fontSize": 22,
              "height": 40,
              "marginBottom": 20,
              "textAlign": "center",
              "transform": Array [
                Object {
                  "translateX": 112.5,
                },
              ],
              "width": 525,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 2,
              "fontSize": 22,
              "height": 40,
              "marginBottom": 20,
              "textAlign": "center",
              "transform": Array [
                Object {
                  "translateX": 112.5,
                },
              ],
              "width": 525,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "black",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 2,
              "elevation": 0,
              "fontSize": 13,
              "left": 375,
              "marginTop": 6.67,
              "minWidth": 64,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 0,
              "transform": Array [
                Object {
                  "translateX": -150,
                },
              ],
              "width": 300,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Helvetica Neue",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "color": "black",
                        "fontFamily": "HelveticaNeue-Medium",
                      },
                      Object {
                        "fontFamily": "HelveticaNeue-Medium",
                      },
                    ],
                  ]
                }
              >
                GITHUB LOGIN
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
